languages/elixir/exercises/concept/processes/.docs/after.md

```diff
@@ -0,0 +1,133 @@
+
+- You can send messages to a process using [send/2][kernel-send].
+
+  ```elixir
+  send(pid, :hello)
+  ```
+
+  - The message ends up in the recipient's _mailbox_.
+  - `send` does not check if the message was received nor if the recipient is still alive.
+
+- A message can be of any type.
+- You can receive a message sent to the current process using [`receive/1`][kernel-receive].
+
+  ```elixir
+  receive do
+    {:ping, sender_pid} -> send(sender_pid, :pong)
+    :do_nothing -> nil
+  end
+  ```
+
+  - You need to pattern match on messages.
+  - `receive` waits until one message matching any given pattern is in the processes mailbox.
+
+    - By default, it waits indefinitely, but can be given a timeout using an `after` block.
+
+      ```elixir
+      receive do
+        {:ping, sender_pid} -> send(sender_pid, :pong)
+      after
+        5000 ->
+          {:error, "No message in 5 seconds"}
+      end
+      ```
+
+  - Read messages are removed from the process' mailbox. Unread messages will stay there indefinitely.
+    - Always write a catch-all `_` clause in `receive/1` to avoid running of out memory due to piled up unread messages.
+
+## Receive loop
+
+If you want to receive more than one message, you need to call `receive/1` recursively. It is a common pattern to implement a recursive function, for example named `loop`, that calls `receive/1`, does something with the message, and then calls itself to wait for more messages. If you need to carry some state from one `receive/1` call to another, you can do it by passing an argument to that `loop` function.
+
+```elixir
+def loop(state) do
+  receive do
+    :increment_by_one ->
+      loop(state + 1)
+
+    {:report_state, sender_pid} ->
+      send(sender_pid, state)
+      loop(state)
+
+    :stop ->
+      nil
+
+    _ ->
+      loop(state)
+  end
+end
+```
+
+In practice, this approach is rarely used directly. As you will learn later, Elixir offers concurrency abstractions, such as the [`Agent` module][agent] or a [_`GenServer` behaviour`_][genserver], that both build on top of the receive loop. However, it is crucial to understand those basics to be able to efficiently use the abstractions.
+
+## PIDs
+
+- PIDs are their own datatype.
+  - You can check if a variable is a PID with [`is_pid/1`][kernel-is-pid]
+- Each process has its own unique PID.
+- You can get the current process' PID with `self()`.
+- PIDs function as _mailbox addresses_ - if you have a process' PID, you can send a message to that process.
+- PIDs are usually created indirectly, as a return value of functions that create new processes, like `spawn`.
+  - PIDs should not be created directly by the programmer, but it is still possible. You can use Erlang's [`list_to_pid/1`][erlang-list-to-pid] function.
+    ```elixir
+    :erlang.list_to_pid('<0.10.0>')
+    # => #PID<0.10.0>
+    ```
```

Comment:
````suggestion
  - PIDs should not be created directly by the programmer. If it were required, Erlang has a [`list_to_pid/1`][erlang-list-to-pid] function.
    ```elixir
    :erlang.list_to_pid('<0.10.0>')
    # => #PID<0.10.0>
    ```
````
We don't have an exercise discussing interacting with erlang functions/modules just yet.  I don't think it's wrong to mention this here, but I think just for our own awareness.